File: 04 Estruturas e Bibliotecas/1068 Balanço de Parênteses I/1068.py
while True:
    try:
        expressao = input()

        aberturas = encerramentos = 0
        expressao_valida = True

        for digito in list(expressao):
            if digito == '(':
                aberturas += 1
            elif digito == ')':
                encerramentos += 1
            else:
                continue

            if encerramentos > aberturas:
                expressao_valida = False

        if expressao_valida and aberturas == encerramentos:
            print('correct')
        else:
            print('incorrect')
    except EOFError:
        break

# https://www.beecrowd.com.br/judge/pt/problems/view/1068
# Ranking 01430º, Time 0.338, Submiss Date 06/03/2023 02:44:00 (Python 3.9)

File: 04 Estruturas e Bibliotecas/1068 Balanço de Parênteses I/1068.c
#include <stdio.h>
#include <string.h>

int main()
{
  char expression[1000];
  while (scanf("%s", expression) != EOF)
  {
    int tam = strlen(expression);
    int as_correct = 1;
    int num_a = 0;
    int num_b = 0;
    for (int i = 0; i < tam; i++)
    {
      if (expression[i] == '(')
      {
        num_a++;
      }
      else if (expression[i] == ')')
      {
        num_b++;
      }
      else
      {
        continue;
      }
      if (num_b > num_a)
      {
        as_correct = 0;
      }
    }

    if ((as_correct == 0) || (num_a != num_b))
    {
      printf("incorrect\n");
    }
    else
    {
      printf("correct\n");
    }
  }
  return 0;
}

// https://www.beecrowd.com.br/judge/pt/problems/view/1022
// Ranking 01170º, Time 0.003, Submiss Date 01/04/2023 14:31:19 (C99)